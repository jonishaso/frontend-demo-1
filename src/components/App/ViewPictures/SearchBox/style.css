.search {
  display: flex;
  flex-flow: row nowrap;
  width: 100%;
}

.search-box {
  font: inherit;
  color: #000;
  width: 100%;
  border-top: none;
  border-right: none;
  border-left: none;
  border-bottom: 2px #3f51b5 solid;
  height: 1.1875em;
  margin: 0;
  display: block;
  padding: 6px 0 7px;
  min-width: 90%;
  background: none;
  animation-name: MuiInputBase-keyframes-auto-fill-cancel;
  -webkit-tap-highlight-color: transparent;
}
.search-box:focus {
  outline: none;
}
