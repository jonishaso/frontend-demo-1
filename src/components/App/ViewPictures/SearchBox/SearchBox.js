import Button from '@material-ui/core/Button'
import FormControl from '@material-ui/core/FormControl'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import { makeStyles } from '@material-ui/core/styles'
import Radio from '@material-ui/core/Radio'
import RadioGroup from '@material-ui/core/RadioGroup'
import ClearIcon from '@material-ui/icons/Clear'
import SearchIcon from '@material-ui/icons/Search'
import React, { useEffect, useState } from 'react'

import './style.css'

const useStyles = makeStyles(theme => ({
  container: {
    display: 'flex',
    flexFlow: 'column wrap',
    width: '90%',
    margin: '20px auto',
    justifyContent: 'space-between',
    alignItems: 'flex-end'
  },
  iconButton: {
    border: 'none'
  },
  formControl: {
    marginBottom: theme.spacing(2),
    marginTop: theme.spacing(2)
  }
}))

const SearchBox = ({ handleSearchClick }) => {
  const classes = useStyles()
  const [keyWords, setKeyWords] = useState('')
  const [imgPerPage, setImgPerPage] = useState(2)

  const cleanKeyWord = () => setKeyWords('')

  const handleKeyWordsChange = event => {
    setKeyWords(event.target.value)
  }

  const handleImgsPerPageChange = event => {
    setImgPerPage(event.target.value)
  }

  const handleSubmit = event => {
    event.preventDefault()
    if (!!keyWords.trim()) {
      handleSearchClick(keyWords, imgPerPage)
    } else {
      alert('keyword could not be empty')
    }
  }

  useEffect(() => {
    if (keyWords.trim() === '') return
    handleSearchClick(keyWords, imgPerPage)
  }, [keyWords, imgPerPage])

  return (
    <form
      onSubmit={handleSubmit}
      className={classes.container}
      noValidate
      autoComplete="off"
    >
      <div className="search">
        <input
          autoFocus
          className="search-box"
          tabIndex="1"
          role="input"
          type="text"
          value={keyWords}
          onChange={handleKeyWordsChange}
        ></input>
        <Button
          className={classes.iconButton}
          onClick={keyWords === '' ? handleSubmit : cleanKeyWord}
          tabIndex="2"
          variant="outlined"
        >
          {(() =>
            keyWords === '' ? (
              <SearchIcon />
            ) : (
              <ClearIcon style={{ color: 'red' }} />
            ))()}
        </Button>
      </div>
      <FormControl component="div" required className={classes.formControl}>
        <RadioGroup
          aria-label=""
          name="imgPerPage"
          value={imgPerPage}
          defaultValue={2}
          onChange={handleImgsPerPageChange}
          row
        >
          <FormControlLabel
            tabIndex="3"
            checked={imgPerPage == 2}
            value={2}
            control={<Radio />}
            label="2"
          />
          <FormControlLabel
            tabIndex="4"
            checked={imgPerPage == 10}
            value={10}
            control={<Radio />}
            label="10"
          />
          <FormControlLabel
            tabIndex="5"
            checked={imgPerPage == 30}
            value={30}
            control={<Radio />}
            label="30"
          />
        </RadioGroup>
      </FormControl>
    </form>
  )
}

export default SearchBox
