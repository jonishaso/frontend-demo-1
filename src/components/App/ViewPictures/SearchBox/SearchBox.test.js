import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SearchBox from './index'

test('test Search Form', () => {
  const wrapper = shallow(<SearchBox />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
