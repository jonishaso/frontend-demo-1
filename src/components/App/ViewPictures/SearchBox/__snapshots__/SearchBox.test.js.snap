// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Search Form 1`] = `
<form
  autoComplete="off"
  className="makeStyles-container-1"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="search"
  >
    <input
      autoFocus={true}
      className="search-box"
      onChange={[Function]}
      role="input"
      tabIndex="1"
      type="text"
      value=""
    />
    <WithStyles(ForwardRef(Button))
      className="makeStyles-iconButton-2"
      onClick={[Function]}
      tabIndex="2"
      variant="outlined"
    >
      <SearchIcon />
    </WithStyles(ForwardRef(Button))>
  </div>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-formControl-3"
    component="div"
    required={true}
  >
    <ForwardRef(RadioGroup)
      aria-label=""
      defaultValue={2}
      name="imgPerPage"
      onChange={[Function]}
      row={true}
      value={2}
    >
      <WithStyles(ForwardRef(FormControlLabel))
        checked={true}
        control={<ForwardRef(WithStyles) />}
        label="2"
        tabIndex="3"
        value={2}
      />
      <WithStyles(ForwardRef(FormControlLabel))
        checked={false}
        control={<ForwardRef(WithStyles) />}
        label="10"
        tabIndex="4"
        value={10}
      />
      <WithStyles(ForwardRef(FormControlLabel))
        checked={false}
        control={<ForwardRef(WithStyles) />}
        label="30"
        tabIndex="5"
        value={30}
      />
    </ForwardRef(RadioGroup)>
  </WithStyles(ForwardRef(FormControl))>
</form>
`;
